Resources/public/js/views/fields/ez-dateandtime-view.js

```diff
@@ -48,24 +48,44 @@ YUI.add('ez-dateandtime-view', function (Y) {
          * @return String
          */
         _formatDate: function (date) {
-            return date.toLocaleDateString();
+            return date.toLocaleDateString(
+                undefined, {year: "numeric", month: "short", day: "numeric"}
+            );
+        },
+
+        /**
+         * Formats the date part of the date object according to the locale
+         * settings of the browser
+         *
+         * @method _formatDateTime
+         * @protected
+         * @param {Date} date
+         * @return String
+         */
+        _formatDateTime: function (date) {
+            var options = {year: "numeric", month: "short", day: "numeric", hour: '2-digit', minute: '2-digit'};
+
+            if ( this.get('fieldDefinition').fieldSettings.useSeconds ) {
+                options.second = '2-digit';
+            }
+            return date.toLocaleTimeString(undefined, options);
         },
 
         /**
          * Formats the time part of the date object according to the locale
          * settings of the browser and to the `useSeconds` field definition
          * settings.
          *
-         * @method _formatDate
+         * @method _formatTime
          * @protected
          * @param {Date} date
          * @return String
          */
         _formatTime: function (date) {
-            var options;
+            var options = {hour: '2-digit', minute: '2-digit'};
```

Comment:
if I'm not mistaken, this method is not used anywhere since it is overridden in ez-time-view but this one is much better so instead of changing it there, you should just deprecate it and move the code based on `toLocaleTimeString` to ez-time-view